Kleinere Optimierungen und neue Methode #CONTAINS_FILES

gehdan opened the pull request:
In diesem Pull-Request sind die Merges von upstream in meine Kopie enthalten und folgende Anpassungen:
- Neue Condition-Methode #CONTAINS_FILES: prüft, ob ein Verzeichnis Dateien enthält
- Anpassung der Condition-Methode #Exists: prüft auch die Existenz von Verzeichnissen
- Optimierung der Beschreibungstexte von <Condition>-Blöcken im Log

gehdan committed:
Merge pull request #3 from FrameworkSystemsGmbH/master

merge upstream

gehdan committed:
Merge pull request #4 from FrameworkSystemsGmbH/master

Bug 2019120851: PBE blockiert FSConsole Log

gehdan committed:
Merge pull request #5 from FrameworkSystemsGmbH/master

Merge upstream

gehdan committed:
Merge pull request #6 from FrameworkSystemsGmbH/master

Merge upstream

gehdan committed:
#Exists prüft auch die Existenz von Verzeichnissen

gehdan committed:
Funktion #CONTAINS_FILES() eingebaut

In einer Bedingung kann im Value-Attribut die Funktion #CONTAINS_FILES()
angegeben werden um zu prüfen, ob ein Verzeichnis Dateien enthält. Das ist v.a.
dann interessant, wenn Build-Ergebnisse später gepackt werden sollen.

Beispiel:
<Condition Name="Prüfen ob Doku rausgeschrieben wurde"
    Value="#CONTAINS_FILES({DocOutputDir})"
    Equals="True">
    <!-- Weitere Anweisungen hier -->
</Condition>

gehdan committed:
Anpassung Beschreibungstext bei <Condition>-Blöcken im Log

Es wird bei Prüf-Methoden in Bedingungen anstelle von "Skip" nun "Execute, late evaluation" im Log ausgegeben. Da in diesem Fall die eigentliche Prüfung erst durchgeführt wird, wenn der Knoten verarbeitet wird.

gehdan commented:
@FSDRE Hier der im Vor-Ort Termin am Do. vorletzte Woche angekündigte Pull request

FSDKO commented:
Vielen Dank für den Pull Request. Wir werden ihn uns bei Zeiten mal anschauen. Er geht nicht unter - versprochen 😉

gehdan committed:
Unterstützung für Parameterüberschreibung

Über den Parameter -p oder --parameter können Variablen an PBE übergeben werden. Sind diese bereits als Parameter in der Konfigurationsdatei gesetzt, so werden diese für den aktuellen Lauf überschrieben. Die Konfigurationsdatei bleibt dabei unangetastet.

gehdan commented:
Ich habe einen weitere Anpassung vorgenommen. Dadurch lassen sich Parameter aus der Konfigurationsdatei über die Befehlszeile überschreiben/setzen. So könnte man z.B. in Batchdateien den Windows-Benutzer an PBE übergeben und in der XML mit dem Parameter arbeiten, obwohl er nicht in der Konfigurationsdatei selbst definiert ist. 

Ein anderes Beispiel ist der SR-Lauf, wie er bei euch vorkommt. Wenn dieser Build auf den Wochentag "Fr" prüft, dieser Build aber nicht lief, oder Probleme verursacht hat. Dann kann man den Parameter Weekday per -p Weekday=Fr über die Befehlszeile setzen und den Build damit auch am Samstag manuell ausführen.

gehdan committed:
Bugfix: CLI-PBE-Parameter werden nicht korrekt verarbeitet

Werden Parameter in anderen Parametern verwendet, so führt eine Parameterüberschreibung per Befehlszeile (CLI) nicht dazu, dass die abhängigen Parameter aktualisiert werden. Daher wurde die Verarbeitung angepasst und der Code etwas aufgeräumt.

Die Verarbeitung der Parameter findet nun im PBEContext statt und die Überschreibung greift direkt bei der Parameterverarbeitung (SetParam)

gehdan committed:
Bugfix: #CONTAINS_FILES() ignoriert Verzeichnisse

gehdan committed:
Merge pull request #7 from FrameworkSystemsGmbH/master

Version 4.6 eröffnen FrameworkSystemsGmbH/fs#534

gehdan commented:
@FSDKO Ich habe einen aktuellen merge der Änderungen auf eurer Seite in meinen fork durchgeführt. Damit ist der Pull-request auf eurer Seite wieder einfach zu mergen.

FSDKO committed:
Merge branch 'master' into pr/gehdan/4

FSDKO committed:
Doku, Unittest, kleinere Fixes

FSDKO committed:
Formatierung

FSDKO committed:
Doku CONTAINS_FILES

FSDKO commented:
@gehdan wir haben endlich etwas Luft, um uns diesem PR zu widmen.

Zu dem Thema mit den Parametern habe ich eine kleine Anpassung vorgenommen. Statt dem `:` als Trenner verwende ich jetzt ein `;`. Mit einem `:` könnte man keine Datei-Pfade angeben. z.B. `--param ExportDir=C:\temp\test`
⚠️ Wenn ihr wieder auf unser Release wechselt (vermutlich mit FS 4.5 im März) dann müsst ihr das ggf. bei euren Scripten anpassen.

Außerdem habe ich jetzt die Doku hier in das Repo gepackt und entsprechend ergänzt. Dann kann das in Zukunft besser laufen 😉.

gehdan commented:
@FSDKO alles klar, top. Das mit den Pfaden macht Sinn 👍
